go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -898,7 +898,13 @@ func (bnc *BaseNetworkController) allocatePodAnnotation(pod *kapi.Pod, existingL
 		return nil, false, fmt.Errorf("cannot retrieve subnet for assigning gateway routes for pod %s, switch: %s",
 			podDesc, switchName)
 	}
-	err = util.AddRoutesGatewayIP(bnc.GetNetInfo(), pod, podAnnotation, network)
+
+	node, err := bnc.watchFactory.GetNode(pod.Spec.NodeName)
+	if err != nil {
+		return nil, false, err
+	}
+
+	err = util.AddRoutesGatewayIP(bnc.GetNetInfo(), node, pod, podAnnotation, network)
```

Comment:
keep me honest here please, this is non IC flow
so atm it is not used, only because https://github.com/ovn-kubernetes/ovn-kubernetes/issues/4876
but the condition that guard about it appears in the function which is used for both IC and non IC 
within `AddRoutesGatewayIP` so it is good to have it as it is now imho